File: master-internship-tracker/scraper/supabase_client.py
import os
from supabase import create_client, Client
from dotenv import load_dotenv

# Load environment variables from .env file if it exists
load_dotenv()

def get_supabase_client() -> Client:
    """
    Initializes and returns a Supabase client using credentials
    from environment variables.
    """
    url = os.environ.get("SUPABASE_URL")
    key = os.environ.get("SUPABASE_KEY")
    
    if not url or not key:
        raise ValueError("Supabase credentials not found. Please set SUPABASE_URL and SUPABASE_KEY environment variables.")
        
    return create_client(url, key)

File: master-internship-tracker/frontend/src/supabaseClient.js
import { createClient } from "@supabase/supabase-js";

const supabaseUrl = "https://ezvnmjnqushpjizcavbe.supabase.co";
const supabaseKey =
  "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImV6dm5tam5xdXNocGppemNhdmJlIiwicm9sZSI6ImFub24iLCJpYXQiOjE3ODc4MDM3MzQsImV4cCI6MjEwMzM3OTczNH0.2w-JobmztNKJgdPPFYCt3j_dpfIF4_Rhy776wWCLpFc";

export const supabase = createClient(supabaseUrl, supabaseKey);
